qiskit/converters/dag_to_circuit.py

```diff
@@ -60,7 +67,10 @@ def dag_to_circuit(dag):
     circuit.calibrations = dag.calibrations
 
     for node in dag.topological_op_nodes():
-        circuit._append(CircuitInstruction(copy.deepcopy(node.op), node.qargs, node.cargs))
+        op = node.op
+        if copy_operations:
+            op = copy.deepcopy(op)
+        circuit._append(CircuitInstruction(op, node.qargs, node.cargs))
```

Comment:
We don't have an optimising compiler to help us here, so it's possible it _might_ be a measurable improvement to move the `if copy_operations` outside the loop, especially in the case where it's false.